docs/reference/performance.rst

```diff
@@ -0,0 +1,277 @@
+.. meta::
+  :description: This chapter describes a set of best practices designed to help
+   developers optimize the performance of HIP-capable GPU architectures.
+  :keywords: AMD, ROCm, HIP, CUDA, performance, guidelines
+
+*******************************************************************************
+Performance Guidelines
+*******************************************************************************
+
+The AMD HIP Performance Guidelines are a set of best practices designed to help
+developers optimize the performance of AMD GPUs. They cover established
+parallelization and optimization techniques, coding metaphors, and idioms that
+can greatly simplify programming for HIP-capable GPU architectures.
+
+By following four main cornerstones, we can exploit the performance 
+optimization potential of HIP.
+
+- parallel execution
+- memory usage optimization
+- optimization for maximum instruction
+- minimizing memory thrashing
+
+In the following chapters, we will show you their benefits and how to use them
+effectively.
+
+.. _parallel execution:
+Parallel execution
+===============================================================================
+For optimal use, the application should reveal and efficiently allocate as much
+parallelism as possible to keep all system components active.
+
+Application level
+-------------------------------------------------------------------------------
+The application should optimize parallel execution across the host and devices 
+using asynchronous calls and streams. Workloads should be assigned based on 
+efficiency: serial to the host, parallel to the devices.
+
+For parallel workloads, when threads need to synchronize to share data, if they
+belong to the same block, they should use `__syncthreads()` (see:
+:ref:`synchronization functions`) within the same kernel invocation. If they 
+belong to different blocks, they must use global memory with two separate 
+kernel invocations. The latter should be minimized as it adds overhead.
```

Comment:
Hmm, e.g. SYCL has a guaranty that lowest global index block that is still executing is progressing. It allows some limited use of synchronization for later blocks in the same invocation. However, I don't remember, if it is the case for HIP, and cannot google it fast. Should we ask Young about?

Reply:
As I know, no. But I do not find it either. It would be a good idea.